update ui

feos7c5 opened the pull request:
Hello, 
here is an update suggestion for the UI. 
I know it's a few deep changes but maybe it makes it easier to use.

<img width="1026" alt="3d" src="https://github.com/user-attachments/assets/fd130351-7964-46cf-94d2-dd32afc42378" />
<img width="1026" alt="1d" src="https://github.com/user-attachments/assets/3ae05b2f-ef2d-4733-beb0-a309ae4343ce" />
<img width="1026" alt="2l" src="https://github.com/user-attachments/assets/4802be46-b56f-4a43-9764-8059690b38fb" />
<img width="1026" alt="1l" src="https://github.com/user-attachments/assets/887874cb-0d45-4478-9ef4-19034728a751" />
<img width="1026" alt="4d" src="https://github.com/user-attachments/assets/1f7d1fc2-7793-4913-af67-e12c803d250d" />
<img width="1026" alt="3l" src="https://github.com/user-attachments/assets/fb669d5b-b694-4439-a809-160d4916c45d" />
<img width="1026" alt="2d" src="https://github.com/user-attachments/assets/5e9df48c-c299-49bb-b5d5-4e548c4a09d9" />
<img width="1026" alt="4l" src="https://github.com/user-attachments/assets/63b1fc41-045c-4a05-8c9c-8178c062f88b" />

glebkudr commented:
Nice design, but I should point to some UX issues about current and future scenarios:

1) The Tree view should always be visible on the 1st **and** 2nd screen, please consider inserting it there. 
2) The next planning thing is "smart context picker" which should also located on the 2nd screen. See https://github.com/glebkudr/shotgun_code/issues/14#issuecomment-2899194438

I would suggest you rethink and support this to be merged into the prooduct.

cristian-milea commented:
Great addition @feos7c5 ! 
I was just thinking about this because of my eyes... for last few nights, looking at "Snow White" Shotgun app :)

kamellperry commented:
@feos7c5 are you planning on making these updates that were requested before merge?

feos7c5 commented:
I have added collapsible sections to the RightSidebar to support the desired functions. 
The FileTree is now available at Step2. The start context picker can be easily added with this structure.

![c2](https://github.com/user-attachments/assets/e727c091-8624-4124-90fa-19bd52489776)
![c1](https://github.com/user-attachments/assets/76b40227-d465-4c5b-b251-4d18de57996c)

kamellperry commented:
This looks really awesome, thank you for taking the time to contribute this awesome UI!

The only thing I would suggest is actually allowing the project files context picker to be open by default, and at the top of the list of collapsible elements (right below mode). Then the token count as something that isn't collapsible. User should always know what's going on there, custom rules at the bottom, "mode" fine where it is.

During my own personal use I find myself changing the files in context often, and while a small change, I believe is a huge UX improvement.

glebkudr commented:
Sorry, very busy at work, I can do this on Sunday. 

@feos7c5 If you still open to modifications, I suggest @kamellperry proposal. Even better, you can make the collapsible options remember their state for future uses (you may use the same save mechanism as it implemented for storing custom rules)

feos7c5 commented:
I have updated the code to reload last stats of collapsible sections and slo declared default as described. 
In addition I added a "StatusBar" with token count. 

![pj](https://github.com/user-attachments/assets/7271e2d1-6e9b-4c44-b359-4b087e9f75fd)

HALDRO commented:
> I have updated the code to reload last stats of collapsible sections and slo declared default as described. In addition I added a "StatusBar" with token count.
The author made the initial display of steps on the top as browser tabs not on the left sidebar for a reason. The left sidebar just for 4 tabs eats up a lot of space and makes other more useful context shrink.

And as for me, the project tree should be more compact, without unnecessary paddings.

I would also consider hot checkboxes for the tree, for example "deselect all", "select all", "do not select files with a dot at the beginning"

Also, in addition to the general token counter, it would be useful to see the number of tokens right in the tree next to each file\folder.

And it is necessary to leave a button for the link to the author's repository :)

feos7c5 commented:
The only reference to my pull request is the navigation. Your other points are additional requirements that are not directly related to my request.

The design itself is an evaluated pattern of known programs and not rocket science. If someone tells me that there is not enough space for the content or that it is more confusing than the original design, then I will question it. 
 
We can talk about further adapting the navigation to a smaller format with icons and numbers, but somewhere the line has to be drawn for this request. I have certainly already invested 3 hours and can use my time elsewhere if new requests come in every 3 messages or everything is called into question.

feos7c5 commented:
The author has already agreed to the design change and the rest of your points have absolutely nothing to do with my pull request.

I am open to constructive suggestions, but this kind of communication is not helpful. If there are constructive suggestions, such as a smaller version of the menu for switching, I would understand. Every 3 comments someone else comes up with new requests. That doesn't have much to do with targeted pull requests.

vailcody commented:
🙋 Guys! Please, add count of lines near to filename. It's convenient to see big files, and ask refactor them.

ubranch commented:
> Please merge this😭 that dark mode is AMAZING _and other things too_

> > I have updated the code to reload last stats of collapsible sections and slo declared default as described. In addition I added a "StatusBar" with token count.
> 
> The author made the initial display of steps on the top as browser tabs not on the left sidebar for a reason. The left sidebar just for 4 tabs eats up a lot of space and makes other more useful context shrink.
> 
> And as for me, the project tree should be more compact, without unnecessary paddings.
> 
> I would also consider hot checkboxes for the tree, for example "deselect all", "select all", "do not select files with a dot at the beginning"
> 
> Also, in addition to the general token counter, it would be useful to see the number of tokens right in the tree next to each file\folder.
> 
> And it is necessary to leave a button for the link to the author's repository :)

<img width="1888" height="971" alt="appimage" src="https://github.com/user-attachments/assets/7863fb10-7110-4e03-8369-6be5f98e9a0b" />


i have one too.
